src/sensitivity_algorithms.jl

```diff
@@ -768,34 +754,33 @@ Base.@pure function NILSS(nseg, nstep; nus=nothing, rng=Xorshifts.Xoroshiro128Pl
   autojacvec=autodiff,
   g=nothing
 )
-  NILSS{chunk_size,autodiff,diff_type,typeof(rng),typeof(nus),typeof(g)}(rng, nseg, nstep, nus, autojacvec, g)
+  NILSS{chunk_size,autodiff,diff_type,typeof(rng),typeof(nus),typeof(g)}(rng,nseg,nstep,nus,autojacvec,g)
 end
 
 """
 NILSAS{CS,AD,FDT,RNG,SENSE,gType} <: AbstractShadowingSensitivityAlgorithm{CS,AD,FDT}
 
 An implementation of the adjoint-mode, continuous
-[non-intrusive adjoint least squares shadowing](https://arxiv.org/abs/1801.08674) method. 
-`NILSAS` allows for computing sensitivities of long-time averaged quantities with respect 
+[non-intrusive adjoint least squares shadowing](https://arxiv.org/abs/1801.08674) method.
+`NILSAS` allows for computing sensitivities of long-time averaged quantities with respect
 to the parameters of an `ODEProblem` by constraining the computation to the unstable subspace.
-`NILSAS` employs SciMLSensitivity.jl's continuous adjoint sensitivity methods on each segment 
-to compute (homogenous and inhomogenous) adjoint solutions. To avoid an exponential blow-up 
-of the adjoint solutions, the trajectory should be divided into sufficiently small segments, 
-where the adjoint solutions are rescaled on the interfaces. The computational and memory cost 
-of NILSAS scale with the number of unstable, adjoint Lyapunov exponents (instead of the number 
-of states as in the LSS method). `NILSAS` avoids the explicit construction of the Jacobian at 
-each time step and thus should generally be preferred (for large system sizes) over `AdjointLSS`. 
-`NILSAS` is favourable over `NILSS` for many parameters because NILSAS computes the gradient 
-with respect to multiple parameters with negligible additional cost. 
+`NILSAS` employs SciMLSensitivity.jl's continuous adjoint sensitivity methods on each segment
+to compute (homogenous and inhomogenous) adjoint solutions. To avoid an exponential blow-up
+of the adjoint solutions, the trajectory should be divided into sufficiently small segments,
+where the adjoint solutions are rescaled on the interfaces. The computational and memory cost
+of NILSAS scale with the number of unstable, adjoint Lyapunov exponents (instead of the number
+of states as in the LSS method). `NILSAS` avoids the explicit construction of the Jacobian at
+each time step and thus should generally be preferred (for large system sizes) over `AdjointLSS`.
+`NILSAS` is favourable over `NILSS` for many parameters because NILSAS computes the gradient
+with respect to multiple parameters with negligible additional cost.
 
 ## Constructor
 
 ```julia
 NILSAS(nseg, nstep, M=nothing; rng = Xorshifts.Xoroshiro128Plus(rand(UInt64)),
-                                adjoint_sensealg = BacksolveAdjoint(),
+                                adjoint_sensealg = BacksolveAdjoint(autojacvec=ReverseDiffVJP()),

```

Comment:
why hardcode reversediffvjp into the defaults here?

Reply:
`NILSAS` does not have a corresponding `adjoint_sensitivities` function yet. So, I think there are three possibilities:
- hardcode it (basically chosen because I just wanted to see if all tests still pass)
- add an algorithm selection within NILSAS 
```julia
adjoint_sensealg = if isinplace(sol.prob)
      setvjp(adjoint_sensealg, inplace_vjp(sol.prob, sol.prob.u0, sol.prob.p, false, false))
    else
      setvjp(adjoint_sensealg, ZygoteVJP())
    end
```
(with the downside that it does not have the fallback to a numerical vjp)
- add a corresponding `adjoint_sensitivities` function dispatch for `NILSAS` with the try and catch functionality. 

Should we go for the last one?

Reply:
Let's go with what we have now then, and open an issue to improve this in the future.